backend/utils/chat_notification.py

```diff
@@ -0,0 +1,218 @@
+from NovuPy.events import Events
+from NovuPy.subscribers import Subscribers
+from utils.room_utils import get_room_members, get_room
+from fastapi import HTTPException
+
+subscriber = Subscribers()
+event = Events()
+
+class Notification:
+    """
+    A class that handles the Novu instance message notification 
+    trigger for Channels, DM and when user is tagged.
+    This class is called after the message instance or object has been created
+    """  
+
+    async def check_message_tag(
+        self, message_obj, payload, 
+        sender_id
+        ):
+        """
+        A method that creates a notification trigger for tagged users
+        Args:
+            message_object-> Dict
+            payload-> Dict: message data that should be passed to novu
+            sender_id -> str
+        Raise:
+            HTTP_422- when Novu couldn't send notification to tagged users
+        """
+        tagged_users = []
+        # user_msg_tag = message_obj["richUiData"]['entityMap']
+        user_msg_tag = message_obj.get("entityMap", " ")
+        if user_msg_tag:
+            for i in range(len(user_msg_tag)):
+                member_id = user_msg_tag[str(i)]['data']['mention']['member_id']
+                if member_id == sender_id:
+                    del i[member_id]
+                tagged_users.append(member_id)
+            dm_trigger_create = await event.trigger(
+                'you-have-been-mentioned',
+                {
+                    "to": tagged_users,
+                    "payload": payload
+                })
+            if dm_trigger_create['statusCode'] !=201:
+                raise HTTPException(
+                    status_code=422, 
+                    detail="Novu couldn't send notifications to tagged users"
+                    )
+            return dm_trigger_create
```

Comment:
`dm_trigger_create`??? Can you explain? the function says `check_message_tag` . What's the return type of this function?

Reply:
fixed @mikengr